hack/generator/pkg/conversions/property_assignment_function.go

```diff
@@ -64,6 +64,8 @@ func NewPropertyAssignmentFromFunction(
 		knownLocals:     astmodel.NewKnownLocalsSet(idFactory),
 	}
 
+	result.knownLocals.Add("source")
```

Comment:
Here too: maybe add a comment that I (@unrepentantgeek) intend to do something that's less hacky here.